### PR TITLE
TCL: Remove TCL depedency from mgmt

### PR DESCRIPTION
**Pull request by chitianhao**
This PR will get rid of all TCL dependency from `mgmt` along with other previous PR's.
Now only `proxy`, which is kinda big, is left to be cleaned up.

`processFormSubmission` and `processFormSubmission_noSubstitute` from `WebMgmtUtils` are removed. They are not used anywhere and @SolidWallOfCode suggests we just remove them.

### CODE REVIEW

File: mgmt/WebMgmtUtils.cc [hunk 1]
```diff
@@ -790,120 +790,6 @@ varType(const char *varName)
...
-
-      ink_hash_table_insert(nameVal, name, value);
-      ats_free(name);
-    }
-  }
-  ats_free(submission_copy);
-
-  return nameVal;
-}
-
```
Comment (shinrich) [inline mgmt/WebMgmtUtils.cc]: What takes over this functionality?

Comment (shinrich) [reply mgmt/WebMgmtUtils.cc]: I guess this was defined only.  Never used?

**Comment by chitianhao** [comment]
I think this PR is ready to ship. Any other concerns? @shinrich @SolidWallOfCode

**Comment by bryancall** [comment]
This PR broke the build on master:
```
In file included from /home/bcall/dev/apache/trafficserver/proxy/http/HttpSM.h:39,
                 from Cache.cc:33:
/home/bcall/dev/apache/trafficserver/proxy/http/remap/UrlRewrite.h:117:5: error: ‘InkHashTable’ does not name a type; did you mean ‘TSHashTable’?
     InkHashTable *hash_lookup;
     ^~~~~~~~~~~~
     TSHashTable
/home/bcall/dev/apache/trafficserver/proxy/http/remap/UrlRewrite.h:141:20: error: ‘InkHashTable’ has not been declared
   bool TableInsert(InkHashTable *h_table, url_mapping *mapping, const char *src_host);
                    ^~~~~~~~~~~~
In file included from /home/bcall/dev/apache/trafficserver/proxy/http/HttpSM.h:39,
                 from Cache.cc:33:
/home/bcall/dev/apache/trafficserver/proxy/http/remap/UrlRewrite.h:198:16: error: expected ‘;’ at end of member declaration
   url_mapping *_tableLookup(InkHashTable *h_table, URL *request_url, int request_port, char *request_host, int request_host_len);
                ^~~~~~~~~~~~
                            ;
/home/bcall/dev/apache/trafficserver/proxy/http/remap/UrlRewrite.h:198:41: error: expected ‘)’ before ‘*’ token
   url_mapping *_tableLookup(InkHashTable *h_table, URL *request_url, int request_port, char *request_host, int request_host_len);
                            ~            ^~
                                         )
/home/bcall/dev/apache/trafficserver/proxy/http/remap/UrlRewrite.h:203:22: error: ‘InkHashTable’ has not been declared
   void _destroyTable(InkHashTable *h_table);
                      ^~~~~~~~~~~~
In file included from /home/bcall/dev/apache/trafficserver/proxy/http/HttpSM.h:39,
                 from Cache.cc:33:
/home/bcall/dev/apache/trafficserver/proxy/http/remap/UrlRewrite.h: In member function ‘bool UrlRewrite::MappingsStore::empty()’:
/home/bcall/dev/apache/trafficserver/proxy/http/remap/UrlRewrite.h:122:16: error: ‘hash_lookup’ was not declared in this scope
       return ((hash_lookup == nullptr) && regex_list.empty());
                ^~~~~~~~~~~
/home/bcall/dev/apache/trafficserver/proxy/http/remap/UrlRewrite.h:122:16: note: suggested alternative: ‘BIO_lookup’
       return ((hash_lookup == nullptr) && regex_list.empty());
                ^~~~~~~~~~~
                BIO_lookup
/home/bcall/dev/apache/trafficserver/proxy/http/remap/UrlRewrite.h: In member function ‘void UrlRewrite::DestroyStore(UrlRewrite::MappingsStore&)’:
/home/bcall/dev/apache/trafficserver/proxy/http/remap/UrlRewrite.h:133:25: error: ‘struct UrlRewrite::MappingsStore’ has no member named ‘hash_lookup’
     _destroyTable(store.hash_lookup);
                         ^~~~~~~~~~~
```

**Comment by zwoop** [comment]
It also looks like it crashes ATS (both _server and _manager).

**Comment by SolidWallOfCode** [comment]
I'll take a look.

**Comment by zwoop** [comment]
See #4501

**Comment by chitianhao** [comment]
Just saw this. I was out yesterday and today. I'll also add an autest for reloading the configuration.

**Comment by bryancall** [comment]
Cherry picked to 8.1.x

**Pull request merged**